test/development/acceptance-app/server-components.test.ts

```diff
@@ -292,6 +292,144 @@ createNextDescribe(
       })
     })
 
+    describe('Class component used in Server Component', () => {
+      it('should show error when Class Component is used', async () => {
+        const { session, cleanup } = await sandbox(
+          next,
+          new Map([
+            [
+              'app/page.js',
+              `
+        import React from 'react'
+        export default class Page extends React.Component {
+          render() {
+            return <p>Hello world</p>
+          }
+        }
+        `,
+            ],
+          ])
+        )
+
+        expect(await session.hasRedbox(true)).toBe(true)
+
+        await check(async () => {
+          expect(await session.getRedboxSource(true)).toContain(
+            'This might be caused by a React Class component being rendered in a server component, React Class components only works in client components. Read more: https://nextjs.org/docs/messages/class-component-in-server-component'
```

Comment:
```suggestion
            'This might be caused by a React Class component being rendered in a server component, React Class components only works in Client Components. Read more: https://nextjs.org/docs/messages/class-component-in-server-component'
```